### DISCUSSION TITLE
Configuring/whitelisting/blacklisting Repositories that Renovate Searches

### DISCUSSION THREAD

**Discussion by DonkeysRock** [renovatebot/renovate#19829] [Suggest an Idea]
I work for a large international financial services corporation and we use an internal server instance of renovate bot to run dependency checks on our GitHub hosted application repositories. 

We use the Gradle build system, and due to the way renovate inserts a task into Gradle at runtime, it then re-uses the Nexus repository definitions we have defined in our application build.gradle files.

Our repository definitions include internal proxies like 'maven-all' which include multiple Nexus staging repos from BUILD (non-prod artifacts) through TEST, RC and PROD (production artifacts).

As a consequence when Renovate searches for internal dependencies from a maven-all proxy it finds both non-prod and production released dependencies. Usually the latest non-prod dependency version will have a higher version number, than the latest production dependency artifact and so gets recommended by renovate as the latest version.

We want to be able to modify this behaviour so we can configure renovate to exclude ANY non-prod dependencies it finds (i.e. from Nexus staging repos BUILD, TEST and RC), and therefore ONLY recommend production dependencies from the Nexus PROD repo. 

I have checked through the extensive Renovate documentation, but have been unable to find reference to any such option for configuration of repos that Renovate will target, or ways to black list or white list repos or artifact paths within the Gradle build context, where renovate effectively re-uses the existing repo definitions in build.gradle.

Is this capability currently supported in someway, or if not, could we start a discussion on how this might be achieved?

**Comment by rarkins** [comment]
You can manually override "discovered" registries by configuring `registryUrls` manually as part of `packageRules`

**Comment by viceice** [comment]
renovate doesn't insert a Gradle task for a long time. which renovate version are you using?

**Comment by DonkeysRock** [comment]
Thanks @rarkins and @viceice for you responses - just checked with the admins and it seems the auto updater hasn't been working for a while - we are on 24.32.2

I've asked them to fix and update and then I will investigate @rarkins configuration suggestion
